### PR TITLE
[DOCS] Document `*` path parameter for security cache APIs

### LINKED ISSUE #70048: [Docs] Add more detail to "name" parameter in Security ClearRoleCache API doc
https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-clear-role-cache.html

This clear role cache page says:

> **name**
>    (string) The name of the role. 

Per [the code](https://github.com/elastic/elasticsearch/blob/a92a647b9f17d1bddf5c707490a19482c273eda3/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/role/RestClearRolesCacheAction.java#L48), this parameter is actually interpreted as an array, with support for `_all` and `*`

e.g.
```
POST /_security/role/role1,role2,role3/_clear_cache
```
```
POST /_security/role/*/_clear_cache
```

### PR DESCRIPTION
**Pull request by jrodewig**
The following security cache APIs support a stand-alone wildcard (`*`) path parameter that clears all caches:
* Clear realm cache
* Clear privileges cache
* Clear roles cache
* Clear API key cache

This PR documents support for that wildcard value.

Closes #70048

### Previews
* Clear realm cache: https://elasticsearch_70081.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-api-clear-cache.html
* Clear privileges cache: https://elasticsearch_70081.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-api-clear-privilege-cache.html
* Clear roles cache: https://elasticsearch_70081.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-api-clear-role-cache.html
* Clear API key cache: https://elasticsearch_70081.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-api-clear-api-key-cache.html

### CODE REVIEW

**Comment by ywangd** [comment]
There was discussion about removing the support for `_all` and in favor of `*`. The plan, as I understand it, is to deprecate in 7.x, support with API compat in 8.x and remove in 9. I wonder whether we should not advertise support for `_all` given the said plan?

**Comment by jrodewig** [comment]
> I wonder whether we should not advertise support for _all given the said plan?

Typically, we would still document the functionality. When deprecation rolled around, we would indicate that in the docs.
However, I went ahead and removed the `_all` references here with c12ca1d30a7837fb7299877127e2ce897265a0c1.

File: x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc [hunk 1]
```diff
@@ -27,8 +27,11 @@ For more information about the native realm, see
 [[security-api-clear-role-cache-path-params]]
 ==== {api-path-parms-title}
 
-`name`::
-  (string) The name of the role. 
+`<roles>`::
+(Required, string)
+Comma-separated list of roles to evict from the native role cache. Supports
+wildcard (`*`) patterns. To evict all roles from the cache, use `*`.
```
Comment (tvernum) [inline x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc:33]:
```suggestion
Comma-separated list of roles to evict from the native role cache.
To evict all roles from the cache, use `*`.
```

We don't actually support wildcards other than `*`

File: x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc [hunk 2]
```diff
@@ -40,3 +43,18 @@ to clear the cache for `my_admin_role`:
 --------------------------------------------------
 POST /_security/role/my_admin_role/_clear_cache
 --------------------------------------------------
+
+You can specify multiple roles as a comma-separated list. Wildcard patterns
+are also supported.
+
+[source,console]
+----
+POST /_security/role/my_admin_role,my_test_role-*/_clear_cache
```
Comment (tvernum) [inline x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc:52]:
```suggestion
POST /_security/role/my_admin_role,my_test_role/_clear_cache
```

File: x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc (hunk 1 above)
Comment (ywangd) [reply x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc]: The success just means the request is executed on the node. But it does not mean the roles are actually cleared. If a role is not found, it just gets silently skipped. Unfortunately, this response does not tell you exactly what roles are cleared or even just how many roles are cleared. We also don't have any API to show you the status (e.g. how many 
entries etc) of the cache. These are things that we should improve upon.

One way you can test it is to turn on trace logging for `authz` and authenticate using a user with the role before and after invocation of the clear-cache API. When the role is indeed successfully cleared, you should see a log line says `Building role from descriptors ... for role names ...` when you try to authenticate afterwards.

Comment (jrodewig) [reply x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc]: Thanks for the clarification @ywangd.

Comment (ywangd) [reply x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc]: FYI, I raised https://github.com/elastic/elasticsearch/issues/70281 for things we discussed in this PR.

Comment (jrodewig) [reply x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc]: Thanks @ywangd. Based on that issue, it looks like several of our API docs could use an update similar to this one. I'll work on expanding the scope of this PR to include those fixes as well. Thanks again!

**Comment by jrodewig** [comment]
@tvernum @tvernum This is ready for another look at your convenience. Thanks for all the feedback so far!

File: x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc [hunk 3]
```diff
@@ -27,8 +27,10 @@ For more information about the native realm, see
 [[security-api-clear-role-cache-path-params]]
 ==== {api-path-parms-title}
 
-`name`::
-  (string) The name of the role. 
+`<roles>`::
+(Required, string)
+Comma-separated list of roles to evict from the native role cache. To evict all
```
Comment (ywangd) [inline x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc:32]: Nit: I don't think we have the phrase "native role cache". There is only just one role cache. So I suggest
```suggestion
Comma-separated list of roles to evict from the role cache. To evict all
```

**Review by ywangd — APPROVED** [review]
LGTM

**Review by tvernum — APPROVED** [review]
I agree with Yang's suggestion about "native role cache", but otherwise LGTM

**Pull request merged**